plugins/in_winevtlog/winevtlog.c

```diff
@@ -0,0 +1,786 @@
+                                  0xffffffff,
+                                  0,
+                                  NULL,
+                                  EvtFormatMessageEvent,
+                                  dwBufferSize,
+                                  pBuffer,
+                                  &dwBufferUsed)) {
+                status = GetLastError();
+                *message_size = dwBufferUsed;
+
+                if (status != ERROR_EVT_UNRESOLVED_VALUE_INSERT) {
+                    switch (status) {
+                        case ERROR_EVT_MESSAGE_NOT_FOUND:
+                        case ERROR_EVT_MESSAGE_ID_NOT_FOUND:
+                        case ERROR_EVT_MESSAGE_LOCALE_NOT_FOUND:
+                        case ERROR_RESOURCE_DATA_NOT_FOUND:
+                        case ERROR_RESOURCE_TYPE_NOT_FOUND:
+                        case ERROR_RESOURCE_NAME_NOT_FOUND:
+                        case ERROR_RESOURCE_LANG_NOT_FOUND:
+                        case ERROR_MUI_FILE_NOT_FOUND:
+                        case ERROR_EVT_UNRESOLVED_PARAMETER_INSERT:
+                        {
+                            if (FormatMessageW(FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM |
+                                               FORMAT_MESSAGE_IGNORE_INSERTS,
+                                               NULL,
+                                               status,
+                                               MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT),
+                                               (WCHAR*)(&lpMsgBuf),
+                                               0,
+                                               NULL) == 0)
+                                FormatMessageW(FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM |
+                                               FORMAT_MESSAGE_IGNORE_INSERTS,
+                                               NULL,
+                                               status,
+                                               MAKELANGID(LANG_ENGLISH, SUBLANG_ENGLISH_US),
+                                               (WCHAR*)(&lpMsgBuf),
+                                               0,
+                                               NULL);
+                            error_message = convert_wstr((WCHAR*)lpMsgBuf, CP_ACP);
+                            flb_error("Failed to get message with %d, err = %s", status, error_message);
+                            flb_free(error_message);
+
+                            wMessage = _wcsdup((WCHAR*)lpMsgBuf);
+                            LocalFree(lpMsgBuf);
+
+                            goto cleanup;
+                        }
+                    }
+
+                    if (status != ERROR_INSUFFICIENT_BUFFER) {
+                        flb_error("failed with %d", status);
+                        goto cleanup;
+                    }
+                }
+            }
+        }
+    }
+
+    wMessage = _wcsdup(pBuffer);
```

Comment:
Do we want to verify if this is NULL?

Reply:
Not really want to verify. 
In this context, almost errors are caught up before reaching here.
For safer operation, we should do.